### DISCUSSION TITLE
How can I import product from strapi to shopify?

### DISCUSSION THREAD

**Discussion by kuttim** [strapi/strapi#6820] [Legacy Topics]
Hi, thanks for the great project.

I'm trying to add products to shopify when they're added to Strapi, currently it works only on POST request, not sure where the product creation code is handled in Strapi - I thought the function would run if new product is created, but I was wrong, do I have to modify internal code?

/api/products/controller/products.js

```
module.exports = {

  async create(ctx) {
    let entity;
    entity = await strapi.services.products.create(ctx.request.body);
    entity.shopify = await shopify.product.create({
        title: ctx.request.body.title,
    });

    return sanitizeEntity(entity, { model: strapi.models.products });
  },
};
```

As previously brought up, it works on POST request, but the function does not run when there's product creation through the strapi UI itself.

**Comment by lauriejim** [comment]
Hello! The code you are updating is the code for the REST API - and I assume you are creating / updating from the admin panel.
I suggest you to place your code in the life cycle function instead of this file.
By using the life cycle function the logic will be applied for both, REST API and admin.
https://strapi.io/documentation/v3.x/concepts/models.html#lifecycle-hooks
